backend/danswer/utils/telemetry.py

```diff
@@ -21,14 +28,43 @@ class RecordType(str, Enum):
     FAILURE = "failure"
 
 
-def get_or_generate_uuid() -> str:
+def _get_or_generate_uuid() -> str:
+    global _CACHED_UUID
+
+    if _CACHED_UUID is not None:
+        return _CACHED_UUID
+
+    kv_store = get_dynamic_config_store()
+
+    try:
+        _CACHED_UUID = cast(str, kv_store.load(KV_CUSTOMER_UUID_KEY))
+    except ConfigNotFoundError:
+        _CACHED_UUID = str(uuid.uuid4())
+        kv_store.store(KV_CUSTOMER_UUID_KEY, _CACHED_UUID, encrypt=True)
+
+    return _CACHED_UUID
+
+
+def _get_or_generate_instance_domain() -> str | None:
+    global _CACHED_INSTANCE_DOMAIN
+
+    if _CACHED_INSTANCE_DOMAIN is not None:
+        return _CACHED_INSTANCE_DOMAIN
+
     kv_store = get_dynamic_config_store()
+
     try:
-        return cast(str, kv_store.load(KV_CUSTOMER_UUID_KEY))
+        _CACHED_INSTANCE_DOMAIN = cast(str, kv_store.load(KV_INSTANCE_DOMAIN_KEY))
     except ConfigNotFoundError:
-        customer_id = str(uuid.uuid4())
-        kv_store.store(KV_CUSTOMER_UUID_KEY, customer_id, encrypt=True)
-        return customer_id
+        with Session(get_sqlalchemy_engine()) as db_session:
+            first_user = db_session.query(User).first()
```

Comment:
The logic to determine the "domain" of the danswer instance is to get the first user and use everything after the @ symbol of their email (E.g. danswer.ai)